File: name_extractor/trigger.ts
import 'dotenv/config';
import { TaskClient, textPart, decodeInlineArtifact } from '@blocks-network/sdk';
import type { ProgressEvent, ArtifactEvent, TerminalEvent } from '@blocks-network/sdk';

/**
 * Trigger a task on the name_extractor agent and print the result.
 * Usage: npx tsx trigger.ts
 */
async function main() {
  const client = await TaskClient.create({
    billingMode: 'free',
    apiKey: process.env.BLOCKS_API_KEY!,
  });

  const session = await client.sendMessage({
    agentName: 'name_extractor',
    requestParts: [textPart('Hello my name is Stephen Blum! My friends names are: Torva, Mohammed, OtherWork, Limon, Ahmed, Bogyatears, Looons, Peace, Rawvee, Vozicom, Snot-123, Inpulsor', 'request')],
  });

  console.log('Task created:', session.taskId);

  session.onProgress((event: ProgressEvent) => {
    console.log('[progress]', event.message ?? event.progress ?? '');
  });
  session.onArtifact(async (event: ArtifactEvent) => {
    const ref = event.artifactRef;
    if (ref.kind === 'inline' && ref.data) {
      const bytes = decodeInlineArtifact(ref);
      console.log('[artifact]', new TextDecoder().decode(bytes));
    } else {
      const downloaded = await session.downloadArtifact(ref);
      console.log('[artifact]', new TextDecoder().decode(downloaded.data));
    }
  });
  session.onTerminal((_event: TerminalEvent) => {
    console.log('[done] Task complete');
    session.close();
    client.destroy();
    process.exit(0);
  });
}

main().catch(console.error);

File: gemma_4_llm/trigger.py
"""Trigger a task on the gemma_4_llm agent and print the result."""

import base64
import threading

from blocks_network import SendMessageRequestPart, create_task_client


def main():
    client = create_task_client()

    session = client.send_message(
        agent_name="gemma_4_llm",
        request_parts=[SendMessageRequestPart(part_id="request", text="Hello from trigger!")],
    )

    print(f"Task created: {session.task_id}")

    done = threading.Event()

    def on_progress(event):
        print("[progress]", event.get("message") or event.get("progress") or "")

    def on_artifact(event):
        ref = event.artifact_ref
        if ref is None:
            print("[artifact]", event.raw)
            return
        if ref.kind == "inline" and ref.data:
            text = base64.b64decode(ref.data).decode()
            print("[artifact]", text)
        else:
            downloaded = session.download_artifact(ref)
            print("[artifact]", downloaded.data.decode())

    def on_terminal(event):
        print("[done] Task complete")
        done.set()

    session.on_progress(on_progress)
    session.on_artifact(on_artifact)
    session.on_terminal(on_terminal)

    done.wait(timeout=60)
    session.close()
    client.destroy()


if __name__ == "__main__":
    main()
